tests/specs/mcd/verification.k

```diff
@@ -97,4 +97,18 @@ module VERIFICATION
     rule M:Memory [ K <- V ] [ K' <- V' ]             => M [ K' <- V' ] [ K <- V ]       requires K' <Int K
     rule M:Memory [ K := BUF:ByteArray ] [ K' <- V' ] => M [ K' <- V' ] [ K := BUF ]     requires K' <Int K orBool K +Int #sizeByteArray(BUF) <=Int K'
 
+    // ### dstoken-transferfrom-fail-rough
+
+    rule chop(N +Int M) <Int N => true requires #rangeUInt(256, N) andBool #rangeUInt(256, M) andBool notBool #rangeUInt(256, N +Int M)
+
+    // ### dsvalue-read-pass
+
+    rule M [ K <- V ] => M requires #lookup(M, K) ==Int V
+
+    // ### cat-ilks-pass
+
+    rule chop(hash2(I1, I2) +Int N) => hash2(I1, I2) +Int N requires 0 <=Int N andBool N <Int 3
```

Comment:
This is the one remaining `hash2 ...` lemmas @daejunpark and @denis-bogdanas . Note that this is needed because the proof access the 0th, 1st, and 2nd elements of a Solidity `struct`, which means it access locations like `#hashedLocation("Solidity", STRUCT_LOCATION) +Int OFFSET`, where `OFFSET` is either 0, 1, or 2. These accesses will realistically never wrap around (which indeed the solidity compiler assumes), but it's hard to express directly as a side-condition on the proof.

Reply:
I suggest just adding a comment stating that this is a consequence of hash2(and keccak) being one-way function over a large range.

Reply:
I don't have a better idea of handing such cases than the presented lemma.